commands/release/release_test.go

```diff
@@ -35,12 +36,95 @@ func TestBreakingChangesSubheaderRegex(t *testing.T) {
 	testRegexPattern(t, "Breaking Changes Subheader", breakingChangesSubheaderRegexStr, validStrings, invalidStrings)
 }
 
-func TestDoBreakingChangesExist(t *testing.T) {
-	noVersion :=
+func Test_parseChangeLogFileNegativeTest(t *testing.T) {
+
+	// test inputs
+	noVersionFound :=
+		`#TBD
+* Something
+* Something else`
+
+	tbdNotPresent :=
+		`
+* Something
+* Something else`
+
+	multipleTBDFound :=
 		`# TBD
 * Something
+# TBD
+* Something else`
+
+	noNewUpdatesForCurrentRelease :=
+		`# TBD
+
+		
+# 0.1.0
 * Something else`
 
+	type args struct {
+		changelogFile string
+	}
+
+	tests := []struct {
+		name     string
+		args     args
+		want     bool
+		wantErr  bool
+		errorMsg string
+	}{
+		{
+			name: "noVersionFound",
+			args: args{
+				changelogFile: noVersionFound,
+			},
+			wantErr:  true,
+			want:     false,
+			errorMsg: "No previous release versions were detected in this changelog",
+		},
+		{
+			name: "tbdNotPresent",
+			args: args{
+				changelogFile: tbdNotPresent,
+			},
+			wantErr:  true,
+			want:     false,
+			errorMsg: "TBD header not found while reading changelog.md",
+		},
+		{
+			name: "multipleTBDFound",
+			args: args{
+				changelogFile: multipleTBDFound,
+			},
+			wantErr:  true,
+			want:     false,
+			errorMsg: fmt.Sprintf("Found more than %d TBD headers", expectedNumTBDHeaderLines),
+		},
+		{
+			name: "noNewUpdatesForCurrentRelease",
+			args: args{
+				changelogFile: noNewUpdatesForCurrentRelease,
+			},
+			wantErr:  true,
+			want:     false,
+			errorMsg: "changelog.md is empty for the current release",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			_, err := parseChangeLogFile([]byte(tt.args.changelogFile))
+			if tt.wantErr {
+				if err == nil {
```

Comment:
You'll actually want to use `require.NotNil` here (it's much nicer to use, and `t.Errorf` counterintuitively actually _doesn't_ stop the entire test)